test/test-jsdomparser.js

```diff
@@ -231,34 +231,43 @@ describe("Test HTML escaping", function() {
 
 describe("Script parsing", function() {
   it("should strip ?-based comments within script tags", function() {
-    var html = '<script><?Silly test <img src="test"></script>';
+    var html = '<script>&lt;?Silly test &lt;img src="test"></script>';
     var doc = new JSDOMParser().parse(html);
     expect(doc.firstChild.tagName).eql("SCRIPT");
-    expect(doc.firstChild.textContent).eql("");
+    expect(doc.firstChild.textContent).eql("<?Silly test <img src=\"test\">");
     expect(doc.firstChild.children.length).eql(0);
     expect(doc.firstChild.childNodes.length).eql(1);
   });
 
   it("should strip !-based comments within script tags", function() {
-    var html = '<script><!--Silly test > <script src="foo.js"></script>--></script>';
+    var html = '<script>&lt;!--Silly test > &lt;script src="foo.js">&lt;/script>--></script>';
```

Comment:
Uh, that is a mistake. Already updated the patch to fix it. Please take a look. Thanks.